### DISCUSSION TITLE
npm install @tailwindcss/forms doesn't work

### DISCUSSION THREAD

**Discussion by kruppy** [tailwindlabs/tailwindcss#4256] [Help] [answered]
Hello all,

I started to use tailwind and am currently facing two issues:

When installing tailwind, I use following installation steps:https://user-images.githubusercontent.com/7460265/116897777-f5ae0a00-ac35-11eb-90ba-09d6aa2146f0.png
But: When trying to mount the css with @tailwind base etc, the vue application doesnt recognizes tailwind and styling is not possible (therefore i use step 6:
image

the generated file is then recognized. Is there a manual how to correctly install tailwind with vue-cli and vue3?

I bought tailwindUI and i try to include a form component. In the tailwind.config.js i add "require('@tailwindcss/forms')" as told. But the forms still look out of place. Do I have to perform any command in order to apply the plugin?

Adding a screenshot of how it looks for me:
https://user-images.githubusercontent.com/7460265/116971501-f5575280-acb9-11eb-9ea7-9cfa81011a0a.png
https://user-images.githubusercontent.com/7460265/116971567-0dc76d00-acba-11eb-98e8-ccba3f87d30c.png

Thank you

**Comment by adamwathan** [comment]
Hey! Can you provide an actual example project so we can see how you have it setup? It looks correct looking at the images so must be something subtle somewhere else.

If you can provide a reproduction, @simonswiss can help you out 👍🏻

**Reply by kruppy** [reply]
Hi @adamwathan, thank you for responding. @simonswiss I added my current learning project here: https://github.com/kruppy/tw_issue. Is this what you meant with reproduction?

BR
Thomas

**Reply by kruppy** [reply]
Hey @adamwathan @simonswiss  , its been a couple of days. Could you check my reproduction?

**Reply by simonswiss** [reply]
Hey @kruppy - sorry about the wait, taking a look at this today 👍

**Comment by simonswiss** [comment, marked as answer]
Hey @kruppy - thanks for sharing your code project.

I think you're using a "stale" CSS file in `static/css/tailwind.css`, which was generated once and is not staying up to date.

How are you generating that file? Did you use the Tailwind CSS CLI for that, as a one-off task? I am not too familiar with `vue-cli-service`, but I don't think that it has anything to do with the generation of that file, so I am assuming it was generated "manually".

Instead of doing this, you should let `vue-cli-service` run your file through PostCSS.

In [your App's `main.js` entry file](https://github.com/kruppy/tw_issue/blob/main/src/main.js#L6), you should import the source CSS file with the `@tailwind` directives (in your case, `static/css/tw.css` instead of that generated CSS file, which is seemingly not updated.

```diff
// src/main.js

import axios from "axios";
- import "./static/css/tailwind.css";
+ import "./static/css/tw.css";
```

You'll also need to create a `postcss.config.js` file in the root of your project, with the following code inside:

```js
// postcss.config.js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
};
```

The purpose of this file is to tell `vue-cil-service` _what_ PostCSS plugins the input CSS should go through. In this case, it will run through `tailwindcss` and `autoprefixer`.

If you do these two things and restart your `npm run serve` command, you should find out that everything works correctly (see screenshot below) 👍

![image](https://user-images.githubusercontent.com/485747/119576249-b79ea300-bdfb-11eb-96be-82357ca6d8f4.png)


Hope it helps!

**Comment by kruppy** [comment]
Thank you @simonswiss , this helped me indeed!!!!

**Reply by simonswiss** [reply]
Happy to hear this was useful. Sorry again for taking so long before getting back to you! 🙏
